Add ParticipantInfoStringsAdditionalPacket

fyfe opened the pull request:
This pr adds ParticipantInfoStringsAdditionalPacket it's only generated during multiplayer races with more than 16 participants (PC only).

Also all string parameters need checked for a null character and everything after the null character needs removed. This is a python unique thing as C/C++ always stop at the null character.

nomad2k committed:
Strip junk from strings after the null character.

nomad2k committed:
Add ParticipantInfoStringsAdditionalPacket

nomad2k committed:
Make data access more consistent.

Old:
    TelemetryPacket.getValue("raceState")
    TelemetryPacket.tyres[0]["tyreTemp"]
    ParticipantInfoStringsPacket.participants["name"]

New:
    TelemetryPacket["raceState"]
    TelemetryPacket["tyres"][0]["tyreTemp"]
    ParticipantInfoStringsPacket["participants"]["name"]

nomad2k committed:
Code clean up.

ghost commented:
I've added two extra  commits to this pr c004d32 is just a minor code cleanup, 398e935 makes accessing the data in the packet classes more consistent.

nomad2k committed:
Test case for ParticipantInfoStringsAdditionalPacket.

nomad2k committed:
Strings are in utf-16 encoding, attempt to encode them to utf-8.

On Python 3 unrecognised characters are escaped, on Python 2 they're
removed. This isn't ideal but it's the only working solution at the
moment. Python 2 and unicode is painful.

ghost commented:
It turns out the strings are utf-16 encoded. On Python 3 it works, on Python 2 it drops the characters that it can't convert to utf-8. This isn't ideal but it's the only working solution at the moment. Python 2 and unicode is painful.
